src/utils/utils_cleaner.c

```diff
@@ -333,8 +332,12 @@ static int _ocf_cleaner_update_metadata(struct ocf_request *req)
 
 		cache_line = iter->coll_idx;
 
+		ocf_hb_cline_prot_lock_wr(&cache->metadata.lock,
+				req->lock_idx, req->map[i].core_id,
+				req->map[i].core_line);
+
 		if (!metadata_test_dirty(cache, cache_line))
-			continue;
+			goto unlock;
```

Comment:
Jumping inside the loop? Maybe invert the logic:
```c
if (metadata_test_dirty(cache, cache_line)) {
        <do_stuff>
}
<unlock>
```

Reply:
ok :)